api/src/main/java/com/datatorrent/api/DAG.java

```diff
@@ -281,6 +306,20 @@
   public abstract OperatorMeta getMeta(Operator operator);
 
   /**
+   * Return all operators present in the DAG.
+   * @param <T>
+   * @return
+   */
+  public <T extends OperatorMeta> Collection<T> getAllOperators();
+
+  /**
+   * Returns all Streams present in the DAG.
+   * @param <T>
+   * @return
+   */
+  public <T extends StreamMeta> Collection<T> getAllStreams();
+
+  /**
```

Comment:
A general comment, I understand these are already public but wonder if the meta objects (other than stream in some way) should be exposed in the api. The users supply concrete operators and port object and the api should probably return those. I understand this is hard to change now without breaking compatibility and probably not advisable now.